### PR TITLE
Added support for AWS inf2 instance types

### LINKED ISSUE #5528: Missing support for AWS inf2 instance types
**Which component are you using?**:

AWS autoscaler

**Is your feature request designed to solve a problem? If so describe the problem this feature should solve.**:

Require support for upcoming inf2 instance types.

**Describe the solution you'd like.**:

Add inf2 instance type definitions to AWS ec2 config.

**Describe any alternative solutions you've considered.**:

None

**Additional context.**:

None

### PR DESCRIPTION
**Pull request by mmcclean-aws**
Added support for the AWS Inferentia 2 instance types based on the NeuronCore v2 chip architecture

#### What type of PR is this?

/kind feature

#### Which issue(s) this PR fixes:

Fixes #5528

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by jbartosik** [comment]
/hold

I don't understand why this PR is changing VPA files. This seems weird because there is a change in a vendored file but there are no corresponding changes in go.mod  / go.sum

**Comment by gjtempleton** [comment]
Agree with @jbartosik .

If you're needing to update the vendored AWS SDK for the CA, you can see instructions on doing that here: https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/aws#using-the-aws-sdk-vendored-in-the-aws-cloudprovider

**Comment by mmcclean-aws** [comment]
thanks @gjtempleton and @jbartosik. Does that mean that in order to add a new instance type this repo doesn't need to be updated ? All you need to do is either wait till the AWS SDK is updated or follow the instructions outlined in the link shared ?

**Comment by x13n** [comment]
IIUC you need to update the sdk based on the instructions from the link. This change should really be limited to changes within `cluster-autoscaler/cloudprovider/aws` dir. Assigning AWS owners for this change.

/assign @gjtempleton @jaypipes @drmorr0 

FWIW, we should really do https://github.com/kubernetes/autoscaler/issues/4878 (and perhaps a similar change for VPA) to limit confusion in the future.

**Comment by gjtempleton** [comment]
There's a couple of bits necessary for this, the update to the [hardcoded list](cluster-autoscaler/cloudprovider/aws/ec2_instance_types.go) which you've already included and following the instructions linked to previously to update the AWS SDK version we vendor in to the AWS cloudprovider implementation for the CA. 

The changes to the VPA related files should be rolled back as @jbartosik and @x13n say.

**Comment by mmcclean-aws** [comment]
I have now rolled back the VPA related files as requested

File: cluster-autoscaler/cloudprovider/aws/ec2_instance_types.go [hunk 1]
```diff
@@ -1614,6 +1614,34 @@ var InstanceTypes = map[string]*InstanceType{
...
+		GPU:          0,
+		Architecture: "amd64",
+	},
+	"inf2.xlarge": {
+		InstanceType: "inf2.xlarge",
+		VCPU:         4,
+		MemoryMb:     16384,
+		GPU:          0,
+		Architecture: "amd64",
+	},
```
Comment (gjtempleton) [inline cluster-autoscaler/cloudprovider/aws/ec2_instance_types.go:1644]: Trailing bit of whitespace is what's causing the failing check here, if we get that fixed up we can get this merged.
```suggestion
	},
```

**Comment by ellistarn** [comment]
Thanks for the help on this @gjtempleton! In the short term, we may need to continue to support ad-hoc manual support for preview EC2 instance types. I'm happy to review and support PRs to the AWS provider for this, if you're comfortable continuing this practice.

**Comment by gjtempleton** [comment]
Thanks.
Removing the hold as now only touching expected CA files.
/unhold
/lgtm
/approve

**Comment by gjtempleton** [comment]
Thanks @ellistarn 
More generally, will keep an eye out for future PRs generating this instance list potentially removing these ad-hoc ones, if it becomes a common occurrence then we may want to look at some method for automating pulling in these instance types which aren't yet publicly listed by the SDK.

**Pull request merged**